libraries/GCS_MAVLink/GCS_Rally.cpp

```diff
@@ -101,3 +102,73 @@ void GCS_MAVLINK::handle_common_rally_message(mavlink_message_t *msg)
     }
 }
 
+MAV_MISSION_RESULT MissionItemProtocol_Rally::get_item(const GCS_MAVLINK &_link,
+                                                       const mavlink_message_t &msg,
+                                                       const mavlink_mission_request_int_t &packet,
+                                                       mavlink_mission_item_int_t &ret_packet)
+{
+    RallyLocation rallypoint;
+
+    if (!rally.get_rally_point_with_index(packet.seq, rallypoint)) {
+        return MAV_MISSION_INVALID_SEQUENCE;
+    }
+
+    ret_packet.command = MAV_CMD_NAV_RALLY_POINT;
+    ret_packet.x = rallypoint.lat;
+    ret_packet.y = rallypoint.lng;
+    ret_packet.z = rallypoint.alt;
+
+    return MAV_MISSION_ACCEPTED;
+}
+
+void MissionItemProtocol_Rally::set_count(const mavlink_mission_count_t &packet)
+{
+    expected_item_count = packet.count;
+}
+void MissionItemProtocol_Rally::timeout()
+{
+    link->send_text(MAV_SEVERITY_WARNING, "Rally upload timeout");
+    expected_item_count = -1;
+}
+
+void MissionItemProtocol_Rally::convert_MISSION_ITEM_INT_to_RallyLocation(const mavlink_mission_item_int_t &cmd, RallyLocation &ret) const
+{
+    ret.lat = cmd.x;
+    ret.lng = cmd.y;
+    ret.alt = constrain_int32(cmd.z, INT16_MIN, INT16_MAX);
```

Comment:
Why constrain here, instead of bounce as out of range?

Reply:
I think I was emulating behaviour elsewhere.

I've changed this to return `INVALID_PARAM7`